packages/cocoon_common/lib/task_status.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:collection/collection.dart';
+
+/// Represents differerent states of a task, or an execution of a build target.
+///
+/// TODO(matanlurey): Finish migrating (https://github.com/flutter/flutter/issues/167284):
+/// - [ ] All usages of `Task.status` should be of type `TaskStatus`
+/// - [ ] Stop implementing `String` and handle conversion elsewherer
+/// - [ ] Replace extension type with an `enum`
+extension type const TaskStatus._(String _schemaValue) implements String {
```

Comment:
That's a mighty fancy enum you got there... I'm guessing its mostly because you want it to a) be String-like and b) access the String like value in a const fashion?

I ever tell you how much I hate dart enums since the language first started?

Reply:
I actually am fine with using an `enum` (and want to end up there) but it's a huge breaking change so it's easier to start with an extension type :)